x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/connector_detail/connector_name_and_description_logic.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { Connector } from '@kbn/search-connectors';
+
+import { Status } from '../../../../../common/types/api';
+
+import { Actions } from '../../../shared/api_logic/create_api_logic';
+import {
+  ConnectorNameAndDescriptionApiLogic,
+  PutConnectorNameAndDescriptionArgs,
+  PutConnectorNameAndDescriptionResponse,
+} from '../../api/connector/update_connector_name_and_description_api_logic';
+
+type NameAndDescription = Partial<Pick<Connector, 'name' | 'description'>>;
+
+type ConnectorNameAndDescriptionActions = Pick<
+  Actions<PutConnectorNameAndDescriptionArgs, PutConnectorNameAndDescriptionResponse>,
+  'makeRequest'
+> & {
+  saveNameAndDescription(nameAndDescription: NameAndDescription): NameAndDescription;
+  setConnector(connector: Connector): Connector;
+};
+
+interface ConnectorNameAndDescriptionValues {
+  connector: Connector | null;
+  isFailed: boolean;
+  isLoading: boolean;
+  isSuccess: boolean;
+  status: Status;
+}
+
+export const ConnectorNameAndDescriptionLogic = kea<
+  MakeLogicType<ConnectorNameAndDescriptionValues, ConnectorNameAndDescriptionActions>
+>({
+  actions: {
+    saveNameAndDescription: (nameAndDescription) => nameAndDescription,
+    setConnector: (connector) => connector,
+  },
+  connect: {
+    actions: [ConnectorNameAndDescriptionApiLogic, ['makeRequest']],
+    values: [ConnectorNameAndDescriptionApiLogic, ['status']],
+  },
+  listeners: ({ actions, values }) => ({
+    saveNameAndDescription: ({ name, description }) => {
+      if (values.connector) {
+        actions.makeRequest({
+          connectorId: values.connector.id,
+          description,
+          indexName: values.connector.index_name || '',
```

Comment:
question: why are we posting the index name here? That seems separate from name and description

Reply:
It is the old API, I will try with the connector API probably will simplify this tomorrow.

Reply:
Plot twist: It was the new API, but the FE call was old.